### ISSUE TITLE
CORS issue with Kirby QL

### ISSUE THREAD

**Issue by sophiamsb** [getkirby/kql#53]
## Description
After Updating Kirby 3.9.7 to Kirby 4.0.0-beta-2 , the request query status responds with error code 401 with message "Unauthenticated". 

**Expected behavior**  
The request should be successful and return status code 200 OK 

**Screenshots**  
Here is my config.php and .htaccess files to document accessibility. It works fine with Kirby 3.9.7

<img width="712" alt="Screenshot 2023-10-24 at 11 44 52" src="https://github.com/getkirby/kirby/assets/39967010/441d6070-1362-44c2-b654-3248b2880d5b">
<img width="882" alt="Screenshot 2023-10-24 at 11 45 26" src="https://github.com/getkirby/kirby/assets/39967010/9e578bfa-89ba-4d59-92df-d771c4b1d3f7">

## To reproduce
1. Replace Kirby folder on the server
2. Go to browser and reach the website url
3. Go to dev tools and check console message 
Result : Error message 
" POST https://my-website.com/api/query 403 (forbidden) " 

## Your setup
**Kirby Version**  
Update from Kirby 3.9.7 to Kirby 4.0.0-beta-2

**Your system (please complete the following information)**
 - Device: Mac Pro
 - OS: macOS Monterey
 - Browser: Chrome/ Safari / Firefox

**Additional context**  
using Kirby QL and React

**Comment by lukasbestle** [comment]
I'm confused. Is it "HTTP 401 Unauthenticated" or "HTTP 403 Forbidden"? Also, have you checked if your custom route to respond to `OPTIONS` requests still works?

**Comment by sophiamsb** [comment]
oh sorry maybe I misunderstood the problem : checking the network activity,
the post request is not working:


[image: Screenshot 2023-10-24 at 13.28.13.png]

[image: Screenshot 2023-10-24 at 13.28.23.png]


Le mar. 24 oct. 2023 à 12:47, Lukas Bestle ***@***.***> a
écrit :

**Comment by sophiamsb** [comment]
Sure !
 
Here are the screenshots:
![Screenshot 2023-10-24 at 13 28 23](https://github.com/getkirby/kql/assets/39967010/a6c36701-3cc4-4bbc-9472-aa82c6f383eb)
![Screenshot 2023-10-24 at 13 28 13](https://github.com/getkirby/kql/assets/39967010/a6af6c06-d5a6-4f39-831a-f57c8a9ac672)

About the custom route responding to OPTIONS requests , where can I check that ?

**Comment by lukasbestle** [comment]
The HTTP 403 response can have many reasons. Could you please post the full response including the response body?

Regarding the `OPTIONS` request: You can test that with the following from the browser console:

```js
await fetch("https://dev.sophiamsaoubi.net/api/query", {method: "OPTIONS"});
```

Alternatively you can use tools like cURL on the command line or (if you prefer GUI tools) there are also HTTP apps to test the request.

**Comment by sophiamsb** [comment]
Here is the full response body:
![Screenshot 2023-10-25 at 13 39 42](https://github.com/getkirby/kql/assets/39967010/8c21eddd-8201-44a6-82e5-dd38456f2f18)

and when I check the  OPTIONS request  I get this:
![Screenshot 2023-10-25 at 13 41 32](https://github.com/getkirby/kql/assets/39967010/b7c57d36-9fca-45e8-ad28-29e1e367f3d4)

But when I do the OPTIONS request with the url of on my current website which works, I get the same output...

**Comment by lukasbestle** [comment]
From your screenshot it's a bit hard to see what's going on with the `OPTIONS` request.

But unrelated to that, the issue you are having with KQL (from the first screenshot) is that the current stable version does not yet support Kirby 4. Could you please try if the changes from this PR fix the issue for you? https://github.com/getkirby/kql/pull/52

**Comment by sophiamsb** [comment]
I tried some stuff but still not working... is it the correct fix ? 
path:`site/plugins/kql/composer.json`
change this :
`"require": {
		"php": ">=8.0.0 <8.3.0",
		"getkirby/cms": ">=3.8.2",
		"getkirby/composer-installer": "^1.2.1"
	},`
to this:
`"require": {
		"php": ">=8.0.0 <8.3.0",
		"getkirby/cms": "^3.9",
		"getkirby/kql": "dev-v4-compat",
		"getkirby/composer-installer": "^1.2.1"
	},`
	
path: `site/plugins/kql/src/Kql/interceptors/cms/Content.php`
changed this: 
`<?php

namespace Kirby\Kql\Interceptors\Cms;

use Kirby\Kql\Interceptor;

class Content extends Interceptor
{`
to this :
`<?php

namespace Kirby\Kql\Interceptors\Cms;

use Kirby\Kql\Interceptors\Content\Content as NewContent;

class Content extends NewContent
{`

**Comment by sophiamsb** [comment]
I tried some stuff but still not working... is it the correct fix ?

path: `site/plugins/kql/composer.json`
change this
```
"require": {
		"php": ">=8.0.0 <8.3.0",
		"getkirby/cms": ">=3.8.2",
		"getkirby/composer-installer": "^1.2.1"
	},
```
to this
```
"require": {
		"php": ">=8.0.0 <8.3.0",
		"getkirby/cms":"^3.9",
		"getkirby/kql": "dev-v4-compat",
		"getkirby/composer-installer": "^1.2.1"
	},
```
path: `site/plugins/kql/src/Kql/interceptors/cms/Content.php`
change this
```
<?php

namespace Kirby\Kql\Interceptors\Cms;

use Kirby\Kql\Interceptor;

class Content extends Interceptor
{
```
to this
```
<?php

namespace Kirby\Kql\Interceptors\Cms;

use Kirby\Kql\Interceptors\Content\Content as NewContent;

class Content extends NewContent
{
```

**Comment by lukasbestle** [comment]
We've released KQL 2.1.0 today, which officially supports Kirby 4. Could you please try updating to that and check if it works for you?

**Comment by sophiamsb** [comment]
I upgraded the KQL and it's working ! Thanks for the support :)
